rust/agents/relayer/src/msg/processor.rs

```diff
@@ -188,14 +147,22 @@ impl MessageProcessor {
                 "Sending message to submitter"
             );
             // Finally, build the submit arg and dispatch it to the submitter.
-            let submit_args = SubmitMessageArgs::new(message, checkpoint, proof);
-            self.tx_msg.send(submit_args)?;
+            let submit_args = SubmitMessageArgs::new(message.clone());
```

Comment:
SubmitMessageArgs no longer necessary, just send messages

Reply:
Actually we should keep this we use the retry counts in both versions of the submitter